From 7942412eecdf8920e381d8281316ec92c1b88f58 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michael=20La=C3=9F?= <lass@mail.uni-paderborn.de>
Date: Thu, 2 Nov 2017 21:16:49 +0100
Subject: [PATCH 4/4] gtx: link against libtinfo if termlib is seperated

If ncurses is built with "./configure --with-termlib=tinfo", gtx fails
to link because of an undefined reference to the LINES symbol which is
then provided by libtinfo.so and not libncurses.so.

If ncurses is present, additionally check whether LINES is provided by
ncurses or tinfo and set $LIB_curses accordingly.

This change is based on a patch provided by Bastian Beischer.

FIXES 134420

Reviewed-on: https://gerrit.openafs.org/12760
Tested-by: BuildBot <buildbot@rampaginggeek.com>
Reviewed-by: Benjamin Kaduk <kaduk@mit.edu>
(cherry picked from commit 311f1d28a2f626350b33ad432e674055b62511bd)

Change-Id: I2f69fe51bbefeeb2a17145a88aa9c891644f2f61
---
 src/cf/curses.m4 | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/cf/curses.m4 b/src/cf/curses.m4
index dc01469f5..261305ce1 100644
--- a/src/cf/curses.m4
+++ b/src/cf/curses.m4
@@ -16,7 +16,10 @@ AC_DEFUN([OPENAFS_CURSES_LIB],
 [AC_CACHE_VAL([openafs_cv_curses_lib],
    [save_LIBS="$LIBS"
     openafs_cv_curses_lib=
-    AC_CHECK_LIB([ncurses], [initscr], [openafs_cv_curses_lib=-lncurses])
+    AC_CHECK_LIB([ncurses], [initscr],
+		 [AC_CHECK_LIB([ncurses], [LINES], [openafs_cv_curses_lib=-lncurses],
+			       [AC_CHECK_LIB([tinfo], [LINES],
+					     [openafs_cv_curses_lib="-lncurses -ltinfo"])])])
     AS_IF([test "x$openafs_cv_curses_lib" = x],
 	  [AC_CHECK_LIB([Hcurses], [initscr], [openafs_cv_curses_lib=-lHcurses])])
     AS_IF([test "x$openafs_cv_curses_lib" = x],
-- 
2.15.1

